About page [WIP]

sahil-shubham opened the pull request:
### This PR is regarding the issue #62 
# Changes
- Added icon in sidebar (bigger screens) & topbar (smaller devices)
- Added subtopics with some text
- Added markdown renderer for changelogs
- updated README.md with changed deployment link

# Work to be done
- [ ] Content Update 
    - FAQs
    - Features
    - About
    - Changelogs
- [x] delocalize markdown file and import it into the about page component
- [x] fix padding issue on the info icon sidebar **(bug)**
- [x] noticed missing sun/moon svg on mobile view **(bug)**

## Screnshot

![Screenshot from 2021-01-25 00-00-32](https://user-images.githubusercontent.com/60931421/105639849-71165780-5ea0-11eb-8d17-c61499abd1a0.png)

sahil-shubham commented:
Apologies for the late PR. I would fix the padding issue (which I cannot figure out till now), the markdown import issue, and the about section content till this Tuesday.

xypnox commented on src/components/sidebar.tsx:
```diff
@@ -26,6 +26,12 @@ export default function Sidebar() {
         </NavLink>
       </div>
 
+      <div className="change-theme">
```
The class name is not related to the icon, we can try a common one, say `icon-links` or `side-links` to be more descriptive and avoid confusion.

sahil-shubham replied:
Yes, that would be better. On it.

xypnox commented on src/views/about.tsx:
```diff
@@ -0,0 +1,34 @@
+
+export const About = () => {
+    return (
+        <div className="about-page">
```
Since the About page details are just regular HTML content, we can keep that as a markdown file as well. Can you try whether we can save a markdown file in `public` and fetch and render it from there?

This will make editing the file easier.

xypnox commented on src/views/about.tsx:
```diff
@@ -0,0 +1,34 @@
+
+            <h2>Let's improve this together</h2>
+            <p>
+                Let us know what you think by talking to us directly through our <a href="#" target="_blank" rel="noreferrer">Slack workspace</a>. 
```
`rel=noreferrer noopener`

xypnox replied:
Link for slack: https://join.slack.com/t/listentothiskgp/shared_invite/zt-l4pt2o1z-~qDftV6hhSmY1V5Y1GsL3g
Use this for now.

This will need to be changed after 30 days. We might use a slack invite service like the one used currently by MetaKGP: https://github.com/rauchg/slackin

Thoughts?
cc. @ghostwriternr @mukul-mehta @parth-paradkar @cipherLord

mukul-mehta replied:
Yeah sounds like a good idea 👍

parth-paradkar replied:
Cool :+1:

xypnox commented on src/views/about.tsx:
```diff
@@ -0,0 +1,34 @@
+            <h2>Let's improve this together</h2>
+            <p>
+                Let us know what you think by talking to us directly through our <a href="#" target="_blank" rel="noreferrer">Slack workspace</a>. 
+                You can also see how all of this comes together and works at the programming level by taking a look at our <a href="#" target="_blank" rel="noreferrer">Github repositories</a>.
```
Add link to Github org here.
https://github.com/lttkgp

xypnox reviewed (changes requested):
Nice, I have left some comments. Resolve the link issue which is breaking the build so we can test it out.

sahil-shubham commented:
Fixing a few more things, would request review again after that.

sahil-shubham commented:
@xypnox I have done the following changes after the previous review - 

- changed className from `change-theme` to `icon-links`
- render the whole about page through markdown rather than just rendering the changelogs
- fixed the links to remove the deployment errors
- fixed the padding issue with the icon in the sidebar

## Things left 
- we should focus on the content of the page now
- the page doesn't look good but rather basic at this point

Please review again, and let me know of any changes. Thank you for your time.

sahil-shubham commented:
I also wanted to know about the solution you were suggesting to import and use the markdown, I searched a lot and used a lot of methods. I am not satisfied with the method I am using at the moment, and would like to know how you would have done the same.
